cmake/compiler_flag_sets/default.cmake

```diff
@@ -53,6 +53,7 @@ if (CMAKE_BUILD_TYPE STREQUAL "Release")
     message(DEBUG "Release build. Setting optimization flags.")
     list(APPEND C_FLAG_SET
                 "-O1"
+                "-fno-delete-null-pointer-checks"
```

Comment:
Otherwise segfault on clang cpp17+ on Release (aka `-O1`) at `include/libcyphal/transport/can/delegate.hpp:178` (b/c above `if (memory_header == nullptr)` was optimized away). In my `canardMemoryAllocate_no_memory` I do emulate "out of memory" condition.